### PR TITLE
CL: Re-design Concentrated Liquidity Module 

### PR DESCRIPTION
**Pull request by mattverse**
Closes: #3243

## What is the purpose of the change
This PR serves the purpose of refactoring and re-designing concentrated liquidity module. 

This includes
- better utilizing `PoolI` in the Concentrated Liquidity module
   - Cdc needed for the pool marshalling unmarshalling in this process
- Abstracting the Concentrated Pool method into a separate package 
- Moving calculation methods to `types` package 

## Testing and Verifying
Verification done via existing testing/

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/concentrated-liquidity/swaps.go [hunk 1]
```diff
@@ -6,6 +6,30 @@ import (
...
+	swapFee sdk.Dec, priceLimit sdk.Dec,
+	poolId uint64) (tokenOut sdk.Coin, err error) {
+	pool, err := k.getPoolbyId(ctx, poolId)
+	if err != nil {
+		return sdk.Coin{}, err
+	}
+	poolTickKVStore := k.GetPoolTickKVStore(ctx, poolId)
+	return pool.SwapOutAmtGivenIn(ctx, poolTickKVStore, tokenIn, tokenOutDenom, swapFee, priceLimit, poolId)
+}
+
```
[the remark this answers is not in the document]
Comment (mattverse) [reply x/concentrated-liquidity/swaps.go]: yeah! moved back to keeper!

File: x/concentrated-liquidity/concentrated-pool/pool.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	return p.Liquidity
+}
+
+func (p *Pool) UpdateLiquidity(newLiquidity sdk.Dec) {
+	p.updateLiquidity(newLiquidity)
+}
+
+func (p *Pool) updateLiquidity(newLiquidity sdk.Dec) {
+	p.Liquidity = p.Liquidity.Add(newLiquidity)
+}
```
[the remark this answers is not in the document]
Comment (mattverse) [reply x/concentrated-liquidity/concentrated-pool/pool.go]: this was an unnecessary abstraction, moved it to a single method!

File: x/concentrated-liquidity/concentrated-pool/pool.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+	fmt "fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/v12/x/concentrated-liquidity/types"
+	gammtypes "github.com/osmosis-labs/osmosis/v12/x/gamm/types"
+)
+
+var (
+	// TODO: remove this check, currently commented out since its blocked on further `PoolI` generalization
```
[the remark this answers is not in the document]
Comment (mattverse) [reply x/concentrated-liquidity/concentrated-pool/pool.go:14]: Thanks for the catch!

File: x/concentrated-liquidity/concentrated-pool/codec.go [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+var (
+	amino = codec.NewLegacyAmino()
+
+	// ModuleCdc references the global x/bank module codec. Note, the codec should
+	// ONLY be used in certain instances of tests and for JSON encoding as Amino is
+	// still used for that purpose.
+	//
+	// The actual codec used for serialization should be provided to x/staking and
+	// defined at the application level.
```
[the remark this answers is not in the document]
Comment (mattverse) [reply x/concentrated-liquidity/concentrated-pool/codec.go]: Removed unnecessary legacy comments!

File: proto/osmosis/concentrated-liquidity/position.proto [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+// until clear steps for migration logic and the unknowns for state breaking are
+// investigated for changing proto package.
+package osmosis.concentratedliquidity.v1beta1;
+
+import "cosmos_proto/cosmos.proto";
+import "gogoproto/gogo.proto";
+
+option go_package = "github.com/osmosis-labs/osmosis/v12/x/concentrated-liquidity/types";
+
+message TickInfo {
```
Comment (p0mvn) [inline proto/osmosis/concentrated-liquidity/position.proto:13]: nit: consider moving `TickInfo` to a separate file as well. The per-file logical grouping with `Position` doesn't really make sense IMO

File: x/concentrated-liquidity/concentrated-pool/codec.go [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+		"osmosis.concentratedliquidity.v1beta1.PoolI",
+		(*types.PoolI)(nil),
+		&Pool{},
+	)
+}
+
+var (
+	amino     = codec.NewLegacyAmino()
+	ModuleCdc = codec.NewAminoCodec(amino)
+)
```
Comment (p0mvn) [inline x/concentrated-liquidity/concentrated-pool/codec.go:27]: nit: let's keep var definitions before function definitions

File: x/concentrated-liquidity/concentrated-pool/pool.go [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+	_ types.PoolI = &Pool{}
+)
+
+func NewConcentratedLiquidityPool(poolId uint64, denom0, denom1 string, currSqrtPrice sdk.Dec, currTick sdk.Int) (Pool, error) {
+	denom0, denom1, err := types.OrderInitialPoolDenoms(denom0, denom1)
+	if err != nil {
+		return Pool{}, err
+	}
+	pool := Pool{
+		// TODO: move gammtypes.NewPoolAddress(poolId) to swaproutertypes
```
Comment (p0mvn) [inline x/concentrated-liquidity/concentrated-pool/pool.go:23]: why? I think it belongs here

File: x/concentrated-liquidity/concentrated-pool/pool.go [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+		CurrentTick:      currTick,
+		Liquidity:        sdk.ZeroDec(),
+		Token0:           denom0,
+		Token1:           denom1,
+	}
+
+	return pool, nil
+}
+
+func (p Pool) GetAddress() sdk.AccAddress {
```
Comment (p0mvn) [inline x/concentrated-liquidity/concentrated-pool/pool.go:36]: godocs for everything please

File: x/concentrated-liquidity/concentrated-pool/pool.go [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+func (p Pool) GetCurrentTick() sdk.Int {
+	return p.CurrentTick
+}
+func (p Pool) GetLiquidity() sdk.Dec {
+	return p.Liquidity
+}
+
+func (p *Pool) UpdateLiquidity(newLiquidity sdk.Dec) {
```
Comment (p0mvn) [inline x/concentrated-liquidity/concentrated-pool/pool.go:99]: Test for this would also be good

File: x/concentrated-liquidity/concentrated-pool/swap.go [hunk 10]
```diff
@@ -0,0 +1,18 @@
+package concentrated_pool
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	gammtypes "github.com/osmosis-labs/osmosis/v12/x/gamm/types"
+)
+
+// ApplySwap.
+func (p *Pool) ApplySwap(ctx sdk.Context, tokenIn sdk.Coin, tokenOut sdk.Coin, poolId uint64, newLiquidity sdk.Dec, newCurrentTick sdk.Int, newCurrentSqrtPrice sdk.Dec) error {
```
Comment (p0mvn) [inline x/concentrated-liquidity/concentrated-pool/swap.go:10]: godoc and test please

File: x/concentrated-liquidity/pool_store.go [hunk 11]
```diff
@@ -1,22 +1,46 @@
...
+	"fmt"
+
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
-	"github.com/osmosis-labs/osmosis/v12/osmoutils"
 	types "github.com/osmosis-labs/osmosis/v12/x/concentrated-liquidity/types"
 )
 
-func (k Keeper) getPoolbyId(ctx sdk.Context, poolId uint64) Pool {
+func (k Keeper) MarshalPool(pool types.PoolI) ([]byte, error) {
```
Comment (p0mvn) [inline x/concentrated-liquidity/pool_store.go:11]: does this need to be exported?

File: x/concentrated-liquidity/pool_store.go [hunk 12]
```diff
@@ -1,22 +1,46 @@
...
+	return k.cdc.MarshalInterface(pool)
+}
+
+func (k Keeper) UnmarshalPool(bz []byte) (types.PoolI, error) {
```
Comment (p0mvn) [inline x/concentrated-liquidity/pool_store.go:15]: ditto

Also, having a marshal - unmarshal unit test would be great

File: x/concentrated-liquidity/pool_store.go [hunk 13]
```diff
@@ -1,22 +1,46 @@
...
+	var acc types.PoolI
+	return acc, k.cdc.UnmarshalInterface(bz, &acc)
+}
+
+func (k Keeper) getPoolbyId(ctx sdk.Context, poolId uint64) (types.PoolI, error) {
```
Comment (p0mvn) [inline x/concentrated-liquidity/pool_store.go:20]:
```suggestion
func (k Keeper) getPoolById(ctx sdk.Context, poolId uint64) (types.PoolI, error) {
```

Comment (p0mvn) [inline x/concentrated-liquidity/pool_store.go:20]: test please

File: x/concentrated-liquidity/pool_store.go [hunk 14]
```diff
@@ -1,22 +1,46 @@
...
+	pool, err := k.UnmarshalPool(bz)
+	if err != nil {
+		return nil, err
+	}
+
+	return pool, nil
 }
 
-func (k Keeper) setPoolById(ctx sdk.Context, poolId uint64, pool Pool) {
+func (k Keeper) setPool(ctx sdk.Context, pool types.PoolI) error {
```
Comment (p0mvn) [inline x/concentrated-liquidity/pool_store.go:37]: let's test this too please if doesn't exist

File: x/concentrated-liquidity/pool_test.go [hunk 15]
```diff
@@ -200,227 +200,46 @@ func (s *KeeperTestSuite) TestCalcOutAmtGivenIn() {
...
-			}
-
-			expectedLiquidity := cl.GetLiquidityFromAmounts(currSqrtPrice, lowerSqrtPrice, upperSqrtPrice, test.poolLiqAmount0, test.poolLiqAmount1)
-			s.Require().Equal(expectedLiquidity.String(), updatedLiquidity.String())
-		})
-
-	}
-}
-
-func (s *KeeperTestSuite) TestSwapOutAmtGivenIn() {
```
Comment (p0mvn) [inline x/concentrated-liquidity/pool_test.go:240]: Why were these removed? Not seeing these in the `concentrated-pool` package

File: x/concentrated-liquidity/concentrated-pool/pool.go [hunk 16]
```diff
@@ -0,0 +1,101 @@
+package concentrated_pool
```
Comment (p0mvn) [inline x/concentrated-liquidity/concentrated-pool/pool.go:1]: let's name this either `concentratedpool` or `concentrated` please

Importing packages with underscore isn't convenient

File: x/concentrated-liquidity/swaps.go [hunk 17]
```diff
@@ -1,34 +1,317 @@
...
+type SwapState struct {
+	amountSpecifiedRemaining sdk.Dec // remaining amount of tokens that need to be bought by the pool
+	amountCalculated         sdk.Dec // amount out
+	sqrtPrice                sdk.Dec // new current price when swap is done
+	tick                     sdk.Int // new tick when swap is done
+	liquidity                sdk.Dec
+}
+
+func (k Keeper) CalcOutAmtGivenIn(ctx sdk.Context,
```
Comment (p0mvn) [inline x/concentrated-liquidity/swaps.go:44]: Was this just moved with no changes?

File: x/concentrated-liquidity/swaps.go [hunk 18]
```diff
@@ -1,34 +1,317 @@
...
+	amt1 := swapState.amountCalculated.TruncateInt()
+
+	tokenIn = sdk.NewCoin(tokenInMin.Denom, amt0)
+	tokenOut = sdk.NewCoin(tokenOutDenom, amt1)
+
+	return tokenIn, tokenOut, swapState.tick, swapState.liquidity, swapState.sqrtPrice, nil
+}
+
+// TODO: implement this
+func (k Keeper) SwapOutAmtGivenIn(ctx sdk.Context, tokenIn sdk.Coin, tokenOutDenom string, swapFee sdk.Dec, priceLimit sdk.Dec, poolId uint64) (tokenOut sdk.Coin, err error) {
```
Comment (p0mvn) [inline x/concentrated-liquidity/swaps.go:160]: Ditto: was this just moved with no changes?

File: x/concentrated-liquidity/types/math.go [hunk 19]
```diff
@@ -1,14 +1,40 @@
...
+package types
 
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/v12/osmomath"
 )
 
+// tickToSqrtPrice takes the tick index and returns the corresponding sqrt of the price
+func TickToSqrtPrice(tickIndex sdk.Int) (sdk.Dec, error) {
```
Comment (p0mvn) [inline x/concentrated-liquidity/types/math.go:10]: Why did we move this here?

These seem to be core math functions that shouldn't be exported outside of the module

File: x/concentrated-liquidity/types/pool.go [hunk 20]
```diff
@@ -0,0 +1,39 @@
+package types
+
+import (
+	proto "github.com/gogo/protobuf/proto"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type PoolI interface {
```
Comment (p0mvn) [inline x/concentrated-liquidity/types/pool.go:9]: Why is this called `PoolI`?

Let's call this `ConcentratedPoolExtension` please according to the spec and component diagram we discussed earlier

File: x/concentrated-liquidity/types/pool.go [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+	GetSwapFee(ctx sdk.Context) sdk.Dec
+	// GetExitFee returns the pool's exit fee, based on the current state.
+	// Pools may choose to make their exit fees dependent upon state.
+	GetExitFee(ctx sdk.Context) sdk.Dec
+	// Returns whether the pool has swaps enabled at the moment
+	IsActive(ctx sdk.Context) bool
+	// GetTotalShares returns the total number of LP shares in the pool
+	GetTotalShares() sdk.Int
+
+	SpotPrice(ctx sdk.Context, baseAssetDenom string, quoteAssetDenom string) (sdk.Dec, error)
```
Comment (p0mvn) [inline x/concentrated-liquidity/types/pool.go:28]:
```suggestion
    gammtypes.PoolI
```

File: proto/osmosis/concentrated-liquidity/position.proto (hunk 5 above)
Comment (mattverse) [reply proto/osmosis/concentrated-liquidity/position.proto:13]: Created a separate file

File: x/concentrated-liquidity/concentrated-pool/codec.go (hunk 6 above)
Comment (mattverse) [reply x/concentrated-liquidity/concentrated-pool/codec.go:27]: I just wanted to keep the same format with all the other codec files + I think this structure has a point since we can see the variables right before init wdyt?

File: x/concentrated-liquidity/concentrated-pool/pool.go (hunk 7 above)
Comment (mattverse) [reply x/concentrated-liquidity/concentrated-pool/pool.go:23]: Isn't the purpose of swaprouter types be the module of orchestration between pool models? 
I think at least we should create a method for newpooAddress within the concentrated liquidity module rather than creating a dependecy flow between concenterated liquidity package and the gamm types package. Wdyt?

File: x/concentrated-liquidity/concentrated-pool/pool.go (hunk 8 above)
Comment (mattverse) [reply x/concentrated-liquidity/concentrated-pool/pool.go:36]: added!

File: x/concentrated-liquidity/concentrated-pool/pool.go [hunk 22]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+func (p Pool) SpotPrice(ctx sdk.Context, baseAssetDenom string, quoteAssetDenom string) (sdk.Dec, error) {
+	// if zero for one, we use the pool curr sqrt price directly.
+	if p.Token0 == baseAssetDenom {
+		return p.CurrentSqrtPrice.Power(2), nil
+	} else { // if not, we calculate the reverse spot price by 1 / currentSqrtPrice^2
+		return sdk.NewDec(1).Quo(p.CurrentSqrtPrice.Power(2)), nil
+	}
+}
```
[the remark this answers is not in the document]
Comment (mattverse) [reply x/concentrated-liquidity/concentrated-pool/pool.go:74]: Added!

File: x/concentrated-liquidity/concentrated-pool/pool.go (hunk 9 above)
Comment (mattverse) [reply x/concentrated-liquidity/concentrated-pool/pool.go:99]: Added tests for this!

File: x/concentrated-liquidity/concentrated-pool/swap.go (hunk 10 above)
Comment (mattverse) [reply x/concentrated-liquidity/concentrated-pool/swap.go:10]: Added!

File: x/concentrated-liquidity/pool_store.go (hunk 11 above)
Comment (mattverse) [reply x/concentrated-liquidity/pool_store.go:11]: I don't see why not, but still made it a private method

File: x/concentrated-liquidity/pool_store.go (hunk 13 above)
Comment (mattverse) [reply x/concentrated-liquidity/pool_store.go:20]: Changed!

File: x/concentrated-liquidity/pool_store.go (hunk 12 above)
Comment (mattverse) [reply x/concentrated-liquidity/pool_store.go:15]: What do you think's a good test for the Marshal and unmarshalling? I don't think they do much + hard to test since we get returned into bytes. 

A test case I can think of Marshalling a pool and then unmarshalling it to see if we get the same result. wdyt?

File: x/concentrated-liquidity/types/math.go (hunk 19 above)
Comment (mattverse) [reply x/concentrated-liquidity/types/math.go:10]: I think for methods that does not use Ctx, there's no need to be inside the (keeper) or the core module, instead can be at types, so that it can be accessed more easily from different packages. What's the reason you think it should be in the core module?

File: x/concentrated-liquidity/pool_store.go (hunk 13 above)
Comment (mattverse) [reply x/concentrated-liquidity/pool_store.go:20]: added!

File: x/concentrated-liquidity/concentrated-pool/pool.go (hunk 16 above)
Comment (mattverse) [reply x/concentrated-liquidity/concentrated-pool/pool.go:1]: Why do you think they are inconvenient?

File: x/concentrated-liquidity/swaps.go (hunk 17 above)
Comment (mattverse) [reply x/concentrated-liquidity/swaps.go:44]: Yes, no change

File: x/concentrated-liquidity/swaps.go (hunk 18 above)
Comment (mattverse) [reply x/concentrated-liquidity/swaps.go:160]: Yes, no change

File: x/concentrated-liquidity/types/pool.go (hunk 20 above)
Comment (mattverse) [reply x/concentrated-liquidity/types/pool.go:9]: Changed!

**Comment by czarcas7ic** [comment]
@mattverse merged main into this so I could utilize your branch for further implementations :)

File: x/concentrated-liquidity/concentrated-pool/pool.go (hunk 16 above)
Comment (p0mvn) [reply x/concentrated-liquidity/concentrated-pool/pool.go:1]: When we import this package, we need to refer to it as `concentrated_pool` .

E.g.: `concentrated_pool.RegisterInterfaces(registry)`

Which is inconsistent with style patterns in Go

File: x/concentrated-liquidity/concentrated-pool/pool.go (hunk 7 above)
Comment (p0mvn) [reply x/concentrated-liquidity/concentrated-pool/pool.go:23]: ahh I see, that makes sense. I thought you meant to move address creation itself there and propagate the address from the swap router.

I'm down for your suggestion

**Comment by p0mvn** [comment]
Merged the latest changes and applied some logic on top since Matt is away till Wednesday.

Changes made on top of the merge:

- moved all core proto state to `internal/model` package to avoid exporting it publicly in types
- moved all core math logic to `internal/math` to avoid it being exported
- type casted to concrete struct type instead of registering codec

**Review by p0mvn — APPROVED** [review]

File: x/concentrated-liquidity/lp.go [hunk 23]
```diff
@@ -36,19 +34,22 @@ func (k Keeper) createPosition(ctx sdk.Context, poolId uint64, owner sdk.AccAddr
...
+	if liquidityDelta.IsZero() {
+		return sdk.Int{}, sdk.Int{}, sdk.Dec{}, fmt.Errorf("liquidity delta zero")
 	}
-	sqrtRatioLowerTick, err := types.TickToSqrtPrice(sdk.NewInt(lowerTick))
+
+	// N.B. we only write cache context if actual amounts
+	// returned are greater than the given minimums.
+	cacheCtx, writeCacheCtx := ctx.CacheContext()
+
+	actualAmount0, actualAmount1, err := k.updatePosition(cacheCtx, poolId, owner, lowerTick, upperTick, liquidityDelta)
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/lp.go:46]: good catch

File: x/concentrated-liquidity/swaps.go [hunk 24]
```diff
@@ -204,20 +250,15 @@ func (k Keeper) SwapOutAmtGivenIn(ctx sdk.Context, tokenIn sdk.Coin, tokenOutDen
 		return sdk.Coin{}, err
 	}
 
-	if calcTokenIn.Amount.GT(tokenIn.Amount) {
-		return sdk.Coin{}, fmt.Errorf("tokenIn calculated is larger than tokenIn provided")
-	}
-
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/swaps.go]: any reason this was removed?

File: x/concentrated-liquidity/swaps.go [hunk 25]
```diff
@@ -315,33 +356,29 @@ func (k Keeper) CalcInAmtGivenOut(ctx sdk.Context, tokenOut sdk.Coin, tokenInDen
...
 
 	return sdk.NewCoin(tokenInDenom, swapState.amountCalculated.RoundInt()), swapState.liquidity, swapState.tick, swapState.sqrtPrice, nil
 }
 
-func (k *Keeper) SwapInAmtGivenOut(ctx sdk.Context, tokenOut sdk.Coin, tokenInDenom string, swapFee sdk.Dec, minPrice, maxPrice sdk.Dec, poolId uint64) (tokenIn sdk.Coin, err error) {
-	tokenInCoin, newLiquidity, newCurrentTick, newCurrentSqrtPrice, err := k.CalcInAmtGivenOut(ctx, tokenOut, tokenInDenom, swapFee, sdk.ZeroDec(), sdk.NewDec(9999999999), poolId)
-	if err != nil {
-		return sdk.Coin{}, err
-	}
-
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/swaps.go]: Why was this removed?

File: x/concentrated-liquidity/tick.go [hunk 26]
```diff
@@ -117,39 +119,12 @@ func (k Keeper) GetTickInfo(ctx sdk.Context, poolId uint64, tickIndex int64) (ti
...
-// TODO: test
-func tickToSqrtPrice(tickIndex sdk.Int) (sdk.Dec, error) {
-	sqrtPrice, err := sdk.NewDecWithPrec(10001, 4).Power(tickIndex.Uint64()).ApproxSqrt()
-	if err != nil {
-		return sdk.Dec{}, err
-	}
-
-	return sqrtPrice, nil
-}
-
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/tick.go:152]: this was just moved right?

File: x/concentrated-liquidity/tick_test.go [hunk 27]
```diff
@@ -120,32 +121,3 @@ func (s *KeeperTestSuite) TestNextInitializedTick() {
...
-		tc := tc
-
-		suite.Run(name, func() {
-			sqrtPrice, err := types.TickToSqrtPrice(tc.tickIndex)
-			suite.Require().NoError(err)
-			suite.Require().Equal(tc.sqrtPriceExpected, sqrtPrice.String())
-
-		})
-	}
-}
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/tick_test.go:157]: this was just moved right?

File: x/concentrated-liquidity/types/codec.go [hunk 28]
```diff
@@ -15,10 +15,6 @@ func RegisterCodec(cdc *codec.LegacyAmino) {
 }
 
 func RegisterInterfaces(registry cdctypes.InterfaceRegistry) {
-	registry.RegisterInterface(
-		"osmosis.concentratedliquidity.v1beta1.PoolI",
-		(*ConcentratedPoolExtension)(nil),
-	)
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/types/codec.go]: for my learning, why do we no longer need to register this in CL? Is it because we never utilize PooI and instead utilize ConcentratedLiquidityExtension? Does it matter that ConcentratedLiquidityExtension imports PoolI? I might be a step ahead though and this might need to get reintroduced again later, just want to see if my understanding was off.

**Review by czarcas7ic — APPROVED** [review]
Additions LGTM, thanks for battling through that 🙏

File: x/concentrated-liquidity/swaps.go (hunk 25 above)
Comment (p0mvn) [reply x/concentrated-liquidity/swaps.go]: addressed

File: x/concentrated-liquidity/swaps.go (hunk 24 above)
Comment (p0mvn) [reply x/concentrated-liquidity/swaps.go]: idk. restored

File: x/concentrated-liquidity/types/codec.go (hunk 28 above)
Comment (p0mvn) [reply x/concentrated-liquidity/types/codec.go]: This is because we don't marshal this as `Any` to write to the store. Instead, we directly type cast and write to store which is cleaner than dealing with "Any" marshaling complexity
